### PR TITLE
Fix ResponseRecorder Flush

### LINKED ISSUE #560: Traefik stuck when used as frontend for a streaming API
Hi, I have  a service that streams potentially large responses, in a streaming fashion. 
When I put traefik (v1.0.1) in front of my service, the streaming API doesn't seems to reply anymore.

My real use case is to do streaming on a SolR server that may have very large result-set.
but this small service (Python/Flask) will help to reproduce the problem easily :

```python
from flask import Response
from flask import Flask
app = Flask(__name__)
  
@app.route('/numbers')
def generate_route():
    def generate():
        a = 0
        while True :
            yield(str(a) + "\n")
            a = a + 1
    return Response(generate(), mimetype='text/plain')

if __name__ == "__main__":
    app.run(host='0.0.0.0', port=8080)
```

A call to this '/numbers' API will generate an infinite stream of numbers.
This command, while doing a broken pipe, will however still reply and print number 0 to 9 :

> curl -s localhost:8080/numbers | head -n 10
0
1
2
3
...

When I'm using traefik as a reverse proxy to redirect call on port 80 to 8080, the '/numbers' API doesn't reply anymore.

> curl -s localhost/numbers | head -n 10

(no response here...)

So is there anything that can be done about that ?
Thanks 

Olivier

### PR DESCRIPTION
**Pull request by emilevauge**
This PR adds `Flush` implementation to `ResponseRecorder` middleware.
Fixes #560

Signed-off-by: Emile Vauge <emile@vauge.com>

### CODE REVIEW

File: middlewares/retry.go [hunk 1]
```diff
@@ -9,6 +9,13 @@ import (
 	"net/http"
 )
 
+var (
+	_ http.ResponseWriter = &ResponseRecorder{}
```
Comment (errm) [inline middlewares/retry.go]: sorry don't understand this bit, I don't think I have seen this in go before... what is it doing?

Comment (emilevauge) [inline middlewares/retry.go]: It ensures that `ResponseRecorder` implements `http.ResponseWriter` :)

Comment (vdemeester) [inline middlewares/retry.go]: There is even better : `_ http.ResponseWriter = (*ResponseRecorder)(nil)`

**Comment by vdemeester** [comment]
@emilevauge #560 is closed, is this still required ?

**Comment by emilevauge** [comment]
@vdemeester yep it is still needed. Maybe we could merge that in the master instead, not sure I want to release 1.0.3 for that...

**Comment by vdemeester** [comment]
@emilevauge yep, let's put this into master :angel:

**Comment by vdemeester** [comment]
(ah there is some conflicts with master :sweat_smile:

**Comment by emilevauge** [comment]
Hum... We have to merge #610 before ^^

**Comment by emilevauge** [comment]
Closing in favor of #776

**Pull request closed without merging**